src/filters/filter_shift.hpp

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+/*
+ * This filter shifts the input in through several outputs
+ */
+#include "filters/filter.hpp"
+#include "lib/macros.hpp"
+
+#include "lib/logging.hpp"
+
+namespace dmxfish::filters {
+
+    COMPILER_SUPRESS("-Weffc++")
+
+    template <typename T>
+    class filter_shift: public filter {
+    private:
+        T* input = nullptr;
+        double* time = nullptr;
+        double* switch_time = nullptr;
+        std::vector<T> values;
+        double last_update;
+    public:
+        filter_shift() : filter() {}
+        virtual ~filter_shift() {}
+
+        virtual void setup_filter(const std::map<std::string, std::string>& configuration, const std::map<std::string, std::string>& initial_parameters, const channel_mapping& input_channels) override {
+            MARK_UNUSED(initial_parameters);
+            if(!input_channels.float_channels.contains("time")) {
+                throw filter_config_exception("Unable to link input of shift filter: channel mapping does not contain channel 'time' of type 'double'. Should come from the only time node.");
```

Comment:
I'd suggest to write a full sentence. `This input should come from the scenes global time node` instead of `Should come from the only time node.`.